mbed_host_tests/__init__.py

```diff
@@ -430,91 +357,126 @@ def execute(self):
 
-            result = None
-            if "host_test_name" in CONFIG:
-                if is_host_test(CONFIG["host_test_name"]):
-                    test_supervisor = get_host_test(CONFIG["host_test_name"])
-                    result = self.run_test(test_supervisor)
-                else:
-                    self.notify("HOST: Error! Unknown host test name '%s' (use 'mbedhtrun --list' to verify)!"% CONFIG["host_test_name"])
-                    self.notify("HOST: Error! You can use switch '-e <dir>' to specify local directory with host tests to load")
-                    self.print_result(self.RESULT_ERROR)
-            else:
-                self.notify("HOST: Error! No host test name defined in preamble")
-                self.print_result(self.RESULT_ERROR)
-
-            if result is not None:
-                self.print_result(result)
-            else:
-                self.notify("HOST: Passive mode...")
-        except Exception, e:
-            print str(e)
+        result = self.run_test()
+        if result == True:
+            self.print_result(self.RESULT_SUCCESS)
+        elif result == False:
+            self.print_result(self.RESULT_FAILURE)
+        elif result is None:
             self.print_result(self.RESULT_ERROR)
 
-    def run_test(self, test_supervisor):
+    def run_test(self):
         result = None
-        try:
-            self.test_supervisor = test_supervisor
-            self.test_supervisor.setup()
-            result = self.test_supervisor.test(self)    # This is blocking, waits for {end}
-        except Exception, e:
-            print str(e)
-            self.print_result(self.RESULT_ERROR)
-        finally:
+        timeout_duration = 10       # Default test case timeout
+        event_queue = Queue()       # Events from DUT to host
+        dut_event_queue = Queue()   # Events from host to DUT {k;v}
+
+        callbacks = {
+            "__notify_prn" : lambda k, v, t : self.logger.prn_inf(v)
+        }
+
+        # if True we will allow host test to consume all events after test is funished
+        callbacks_consume = True
+        self.test_supervisor = None
+
+        config = {
+            "digest" : "serial",
+            "port" : self.mbed.port,
+            "baudrate" : self.mbed.serial_baud,
+            "program_cycle_s" : self.options.program_cycle_s
+            }
+
+        self.logger.prn_inf("starting host test process...")
+        start_time = time()
+        args = (event_queue, dut_event_queue, self.prn_lock, config)
+        p = Process(target=conn_process, args=args)
+        p.deamon = True
+        p.start()
+
+        consume_preamble_events = True
+        while (time() - start_time) < timeout_duration:
+            # Handle default events like timeout, host_test_name, ...
+            if event_queue.qsize():
+                (key, value, timestamp) = event_queue.get()
+
+                if consume_preamble_events:
+                    if key == 'timeout':
```

Comment:
Where are these keys described? I see timeout, host_test_name, sync. Are they coming from the client side? should they be defined here as well in this code base? If I find this print "sync KV found" , I have no clue what it means from reading this code